pkg/kv/kvserver/rangefeed/budget.go

```diff
@@ -293,9 +293,15 @@ func (f *BudgetFactory) CreateBudget(key roachpb.RKey) *FeedBudget {
 		return nil
 	}
 	// We use any table with reserved ID in system tenant as system case.
-	if key.Less(roachpb.RKey(keys.SystemSQLCodec.TablePrefix(keys.MaxReservedDescID + 1))) {
-		acc := f.systemFeedBytesMon.MakeBoundAccount()
-		return NewFeedBudget(&acc, 0)
+	if _, id, err := keys.DecodeTenantPrefix(key.AsRawKey()); err == nil {
+		codec := keys.SystemSQLCodec
+		if !id.IsSystem() {
+			codec = keys.MakeSQLCodec(id)
+		}
```

Comment:
nit: I think this branching is unnecessary, since `MakeSQLCodec` will generate an appropriate codec anyway.

Reply:
Isn't that better for performance as SQLCodec is statically generated and will avoid allocating and copying the data locally?

Reply:
Doesn't matter here I think, since this is only called when setting up a new rangefeed? But not important, your call.

Reply:
Doesn't matter now since we can look on the spanconfig instead of trying to decode tenant id. I'll push the change once the test passes stress.